### PR TITLE
[Connector-V2] Add Email sink connector

### PR DESCRIPTION
**Pull request by 2013650523**
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by Hisoka-X** [comment]
Hi, please resolve ci error first. Thanks

**Comment by CalvinKirs** [comment]
If any new Jar binary package adding in your PR, please add License Notice according
[New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)

File: seatunnel-connectors-v2/connector-email/src/main/java/org/apache/seatunnel/connectors/seatunnel/email/sink/EmailSinkWriter.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        stringBuffer.append("\n");
+
+    }
+
+    @Override
+    public void close() {
+        createFile();
+        Properties properties = new Properties();
+
+        properties.setProperty("mail.host", "smtp.qq.com");
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-email/src/main/java/org/apache/seatunnel/connectors/seatunnel/email/sink/EmailSinkWriter.java]: This parameter should be handed over to user control

File: seatunnel-connectors-v2/connector-email/src/main/java/org/apache/seatunnel/connectors/seatunnel/email/sink/EmailSinkWriter.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+        properties.setProperty("mail.transport.protocol", "smtp");
+
+        properties.setProperty("mail.smtp.auth", "true");
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-email/src/main/java/org/apache/seatunnel/connectors/seatunnel/email/sink/EmailSinkWriter.java]: Same problem as above, and these methods may not be appropriate in `close` method

File: seatunnel-connectors-v2/connector-email/src/main/java/org/apache/seatunnel/connectors/seatunnel/email/sink/EmailSinkWriter.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    }
+
+    @Override
+    public void write(SeaTunnelRow element) {
+        Object[] fields = element.getFields();
+
+        for (Object field : fields) {
+            stringBuffer.append(field.toString() + ",");
+        }
+        stringBuffer.deleteCharAt(fields.length - 1);
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-email/src/main/java/org/apache/seatunnel/connectors/seatunnel/email/sink/EmailSinkWriter.java:72]: just as a suggestion. (You can improve it in the next PR, we need to consider the format of various formats, for email)

Comment (2013650523) [reply seatunnel-connectors-v2/connector-email/src/main/java/org/apache/seatunnel/connectors/seatunnel/email/sink/EmailSinkWriter.java:72]: ok,What file formats would you like to have in the future? PDF?

**Comment by CalvinKirs** [comment]
Hi, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

**Comment by CalvinKirs** [comment]
Hi, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

**Comment by 2013650523** [comment]
> Hi, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

done;

File: docs/en/connector-v2/source/Email.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
+# Email
+
+## Description
+
+Send the data as a file to email.
+
+ The tested kudu version is 1.5.6.
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/Email.md]: Please remove this.

File: docs/en/connector-v2/source/Email.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+```bash
+
+ EmailSink {
+      email_from_address = "xxxxxx@qq.com"
+      email_to_address = "xxxxxx@163.com"
+      email_host="smtp.qq.com"
+      email_transport_protocol="smtp"
+      email_smtp_auth="true"
+      email_authorization_code=""
+      email_message_headline="这个是标题"
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/Email.md]: Please use Englinsh in content

Comment (2013650523) [reply docs/en/connector-v2/source/Email.md]: Done;

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**